### ISSUE TITLE
Notification fails to try to send a message if referencing a tag that does not exist

### ISSUE THREAD

**Issue by nisley** [influxdata/influxdb#15269]
I created two checks - one with a Location tag and one without:

![image](https://user-images.githubusercontent.com/1389799/65560321-03949880-def3-11e9-8f28-d2986bc94404.png)

Trying to demonstrate a catch-all Notification Rule, but I believe it is swallowed up even though the alert went CRIT because the message references a Tag that does not exist on the check.

![image](https://user-images.githubusercontent.com/1389799/65560413-71d95b00-def3-11e9-82e2-a8c0c941c506.png)

![image](https://user-images.githubusercontent.com/1389799/65560390-55d5b980-def3-11e9-97c4-bda620f4e62a.png)

**Comment by russorat** [comment]
Sounds like a few issues here:
1. We have no way in the UI to validate that the template message is valid. We could possibly submit a dummy flux query to do that?
2. We are allowed to create a task (check) with an invalid template message. This would need validation from the tasks team i think?
3. We have no way to return errors in the UI. The task team is working on returning errors in the list tasks calls, which we could then display in the UI.

**Comment by desa** [comment]
> We have no way in the UI to validate that the template message is valid. We could possibly submit a dummy flux query to do that?

I don't think that would be sufficient. What if the tag was there, but is no longer. Possibly an alternative would be for flux to return `undefined` or `nil` or something like if a column doesn't exist on a row.

> We are allowed to create a task (check) with an invalid template message. This would need validation from the tasks team i think?

Potentially, I don't know that this is the functionality we want though. If so, I'd imagine it'd be a flux feature.

> We have no way to return errors in the UI. The task team is working on returning errors in the list tasks calls, which we could then display in the UI

👍 


I think my take away from this issue is that
1. We shouldn't let users free-form type templated messages.
2. Flux should handle undefined states more gracefully

**Comment by pauldix** [comment]
It would be nice to have string interpolation that ignores errors and just inserts `r.foo undefined` or whatever, if you're referencing a column that doesn't exist.

But yes to all the stuff about validating what people are putting in. We absolutely should let people put in free form templated messages. We just need a simple way to validate the format. Validating the fields is a separate thing, and as @desa said, it could be valid now but invalid later or something like that.

Ideally, the template message area would have an example `r` from the real data (in check message) or a real status (in notification message) so that a user can go off that. Even better to have it autocomplete in the UI when the user types `${r.`

**Comment by nisley** [comment]
The biggest issue in monitoring in alerting is the notification was swallowed up.

In the case that the check had all the tags, everything worked great and lulled the user into thinking everything was working as expected.

When the second check went to crit, there is nothing in the rule history to indicate the conditions to trigger it were met (no log of a trigger) and of course, no actual notification sent.

**Comment by russorat** [comment]
bump

**Comment by desa** [comment]
Just wanted to note that this also occurs in checks fairly frequently

**Comment by wolffcm** [comment]
I recently hit this issue when writing a check.  It would be nice if we could preview the message in the UI and see if it has any issues. 

The data seems to be pivoted, so one should use `"${r.fieldname}"` to reference fields, rather than `"${r._value}"`.

To make things more interesting, Flux expects the expressions in interpolated strings to be of type string, so `"${r.num_field}"` will fail, but that can be fixed by doing `"${string(v: r.num_field)}"`. I believe there are plans to make Flux relax this requirement, so any value that can be represented as string is supported.

**Comment by wolffcm** [comment]
Issue to allow arbitrary expressions in interpolated strings in Flux: https://github.com/influxdata/flux/issues/1775
